cmd/inspect/block.go

```diff
@@ -0,0 +1,219 @@
+				txIndex = -1
+				if err = txHash.UnmarshalHex(txIndexOrHash); err != nil {
+					cobra.CheckErr(fmt.Errorf("malformed tx hash: %w", err))
+				}
+			}
+		}
+
+		// Establish connection with the target network.
+		ctx := context.Background()
+		conn, err := connection.Connect(ctx, npa.Network)
+		cobra.CheckErr(err)
+
+		fmt.Printf("Network:        %s", npa.NetworkName)
+		if len(npa.Network.Description) > 0 {
+			fmt.Printf(" (%s)", npa.Network.Description)
+		}
+		fmt.Println()
+
+		switch npa.ParaTime {
+		case nil:
+			// Consensus layer.
+			cobra.CheckErr("inspecting consensus layer blocks not yet supported") // TODO
+		default:
+			// Runtime layer.
+			rt := conn.Runtime(npa.ParaTime)
+
+			blk, err := rt.GetBlock(ctx, blkNum)
+			cobra.CheckErr(err)
+
+			fmt.Printf("ParaTime:       %s", npa.ParaTimeName)
+			if len(npa.ParaTime.Description) > 0 {
+				fmt.Printf(" (%s)", npa.ParaTime.Description)
+			}
+			fmt.Println()
+			fmt.Printf("Round:          %d\n", blk.Header.Round)
+			fmt.Printf("Version:        %d\n", blk.Header.Version)
+			fmt.Printf("Namespace:      %s\n", blk.Header.Namespace)
+
+			// TODO: Fix when timestamp has a String method.
+			ts, _ := blk.Header.Timestamp.MarshalText()
+			fmt.Printf("Timestamp:      %s\n", string(ts))
+
+			// TODO: Fix when type has a String method.
+			fmt.Printf("Type:           %d\n", blk.Header.HeaderType)
+			fmt.Printf("Previous:       %s\n", blk.Header.PreviousHash)
+			fmt.Printf("I/O root:       %s\n", blk.Header.IORoot)
+			fmt.Printf("State root:     %s\n", blk.Header.StateRoot)
+			fmt.Printf("Messages (out): %s\n", blk.Header.MessagesHash)
+			fmt.Printf("Messages (in):  %s\n", blk.Header.InMessagesHash)
+
+			txs, err := rt.GetTransactionsWithResults(ctx, blk.Header.Round)
+			cobra.CheckErr(err)
+
+			fmt.Printf("Transacations:  %d\n", len(txs))
+
+			if len(args) >= 2 {
+				fmt.Println()
+
+				// Resolve transaction index if needed.
+				if txIndex == -1 {
+					for i, tx := range txs {
+						if h := tx.Tx.Hash(); h.Equal(&txHash) {
+							txIndex = i
+							break
+						}
+					}
+
+					if txIndex == -1 {
+						cobra.CheckErr(fmt.Errorf("failed to find transaction with hash %s", txHash))
+					}
+				}
+
+				if txIndex >= len(txs) {
+					cobra.CheckErr(fmt.Errorf("transaction index %d is out of range", txIndex))
+				}
+				tx := txs[txIndex]
+
+				fmt.Printf("=== Transaction %d ===\n", txIndex)
+
+				if len(tx.Tx.AuthProofs) == 1 && tx.Tx.AuthProofs[0].Module != "" {
+					// Module-specific transaction encoding scheme.
+					scheme := tx.Tx.AuthProofs[0].Module
+
+					switch scheme {
+					case "evm.ethereum.v0":
+						// Ethereum transaction encoding.
+						var ethTx ethTypes.Transaction
+						if err := ethTx.UnmarshalBinary(tx.Tx.Body); err != nil {
+							fmt.Printf("[malformed 'evm.ethereum.v0' transaction: %s]\n", err)
+							break
+						}
+
+						fmt.Printf("Kind:      evm.ethereum.v0\n")
+						fmt.Printf("Hash:      %s\n", ethTx.Hash())
```

Comment:
Similar to the "Address" vs "Ethereum address", should we discriminate between oasis and eth hash?

```suggestion
						fmt.Printf("Ethereum hash:      %s\n", ethTx.Hash())
						fmt.Printf("Hash:      %s\n", tx.Tx.Hash())
```